fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java

```diff
@@ -223,17 +270,31 @@ private TableDescriptor applySystemDefaults(TableDescriptor tableDescriptor) {
 
     @Override
     public CompletableFuture<DropTableResponse> dropTable(DropTableRequest request) {
+        TablePath tablePath = toTablePath(request.getTablePath());
+        if (authorizer != null) {
+            authorizer.authorize(
+                    currentSession(),
+                    OperationType.DROP,
+                    Resource.table(tablePath.getDatabaseName(), tablePath.getTableName()));
+        }
+
         DropTableResponse response = new DropTableResponse();
-        metadataManager.dropTable(
-                toTablePath(request.getTablePath()), request.isIgnoreIfNotExists());
+        metadataManager.dropTable(tablePath, request.isIgnoreIfNotExists());
         return CompletableFuture.completedFuture(response);
     }
 
     @Override
     public CompletableFuture<CreatePartitionResponse> createPartition(
             CreatePartitionRequest request) {
-        CreatePartitionResponse response = new CreatePartitionResponse();
         TablePath tablePath = toTablePath(request.getTablePath());
+        if (authorizer != null) {
+            authorizer.authorize(
+                    currentSession(),
+                    OperationType.CREATE,
```

Comment:
For partitions, I think we should use WRITE permission under the table. That means if a user has the write permission of a table, then he can create partitions. Otherwise, he has to apply CREATE/DROP permission which is very hard to use. Partition is just an internal data distribution. 

Add IT case for this in `FlinkAuthorizationITCase`